Crash running tlsx in auto-mode

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/tlsx/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #tlsx channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### tlsx version:
<!-- You can find current version of tlsx with "tlsx -version" -->
<!-- We only accept issues that are reproducible on the latest version of tlsx. -->
<!-- You can find the latest version of project at https://github.com/projectdiscovery/tlsx/releases/ -->
dev

### Current Behavior:
<!-- A concise description of what you're experiencing. -->
Crash with auto mode

### Expected Behavior:
<!-- A concise description of what you expected to happen. -->
Scan finishes succesfully

```console
panic: runtime error: invalid memory address or nil pointer dereference\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc8f258]
goroutine 313 [running]:\ngithub.com/projectdiscovery/tlsx/pkg/tlsx/ztls.(*Client).ConnectWithOptions(0xc001216d98, {0xc003e64b90, 0xc}, {0x0, 0x0}, {0xf2041f, 0x3}, {{0x0?, 0xc8b2b2?}})
/root/tlsx/pkg/tlsx/ztls/ztls.go:187 +0x878\ngithub.com/projectdiscovery/tlsx/pkg/tlsx/auto.(*Client).ConnectWithOptions(0xc001216db0, {0xc003e64b90, 0xc}, {0x0, 0x0}, {0xf2041f, 0x3}, {{0x0?, 0xc00001fa98?}})
/root/tlsx/pkg/tlsx/auto/auto.go:42 +0xfd\ngithub.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).ConnectWithOptions(0xc001216d68, {0xc003e64b90, 0xc}, {0x0, 0x0}, {0xf2041f, 0x3}, {{0x0?, 0x0?}})
/root/tlsx/pkg/tlsx/tlsx.go:58 +0xfd\ngithub.com/projectdiscovery/tlsx/internal/runner.(*Runner).processInputElementWorker(0xc000016600, 0x0?, 0x0?)
/root/tlsx/internal/runner/runner.go:152 +0x445\ncreated by github.com/projectdiscovery/tlsx/internal/runner.(*Runner).Execute
/root/tlsx/internal/runner/runner.go:118 +0x65
```

### Anything else:
<!-- Links? References? Screnshots? Anything that will give us more context about the issue that you are encountering! -->

```
{"ip":"104.248.121.26","level":"error","msg":"Could not execute work module tlsx 2.16.0.0/12: could not run tlsx: could not wait for command to finish: exit status 2 =\u003e   \n\n  _____ _    _____  __\n |_   _| |  / __\\ \\/ /\n   | | | |__\\__ \\\u003e  \u003c \n   |_| |____|___/_/\\_\\\tv0.0.5\n\n\t\tprojectdiscovery.io\n\n[\u001b[34mWRN\u001b[0m] Use with caution. You are responsible for your actions.\n[\u001b[34mWRN\u001b[0m] Developers assume no liability and are not responsible for any misuse or damage.\npanic: runtime error: invalid memory address or nil pointer dereference\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc8f258]\n\ngoroutine 313 [running]:\ngithub.com/projectdiscovery/tlsx/pkg/tlsx/ztls.(*Client).ConnectWithOptions(0xc001216d98, {0xc003e64b90, 0xc}, {0x0, 0x0}, {0xf2041f, 0x3}, {{0x0?, 0xc8b2b2?}})\n\t/root/tlsx/pkg/tlsx/ztls/ztls.go:187 +0x878\ngithub.com/projectdiscovery/tlsx/pkg/tlsx/auto.(*Client).ConnectWithOptions(0xc001216db0, {0xc003e64b90, 0xc}, {0x0, 0x0}, {0xf2041f, 0x3}, {{0x0?, 0xc00001fa98?}})\n\t/root/tlsx/pkg/tlsx/auto/auto.go:42 +0xfd\ngithub.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).ConnectWithOptions(0xc001216d68, {0xc003e64b90, 0xc}, {0x0, 0x0}, {0xf2041f, 0x3}, {{0x0?, 0x0?}})\n\t/root/tlsx/pkg/tlsx/tlsx.go:58 +0xfd\ngithub.com/projectdiscovery/tlsx/internal/runner.(*Runner).processInputElementWorker(0xc000016600, 0x0?, 0x0?)\n\t/root/tlsx/internal/runner/runner.go:152 +0x445\ncreated by github.com/projectdiscovery/tlsx/internal/runner.(*Runner).Execute\n\t/root/tlsx/internal/runner/runner.go:118 +0x65\n\n","taskid":"cbp8t778j1tdoqpk4mag","time":"2022-08-09T16:49:04Z"}
```

--- a/pkg/tlsx/auto/auto.go
+++ b/pkg/tlsx/auto/auto.go
@@ -42,6 +42,9 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 		ztlsResponse, ztlsErr := c.ztlsClient.ConnectWithOptions(hostname, ip, port, options)
 		if ztlsErr != nil {
 			opensslResponse, opensslError := c.opensslClient.ConnectWithOptions(hostname, ip, port, options)
+			if opensslError == openssl.ErrNotSupported {
+				return nil, ztlsErr
+			}
 			if opensslError != nil {
 				return nil, opensslError
 			}

--- a/pkg/tlsx/auto/auto.go
+++ b/pkg/tlsx/auto/auto.go
@@ -42,7 +42,7 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 		ztlsResponse, ztlsErr := c.ztlsClient.ConnectWithOptions(hostname, ip, port, options)
 		if ztlsErr != nil {
 			opensslResponse, opensslError := c.opensslClient.ConnectWithOptions(hostname, ip, port, options)
-			if opensslError == openssl.ErrNotSupported {
+			if errors.Is(opensslError, openssl.ErrNotSupported) {
 				return nil, ztlsErr
 			}
 			if opensslError != nil {

--- a/pkg/tlsx/tls/tls.go
+++ b/pkg/tlsx/tls/tls.go
@@ -113,6 +113,9 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 	if err != nil {
 		return nil, errors.Wrap(err, "could not dial address")
 	}
+	if rawConn == nil {
+		return nil, fmt.Errorf("could not connect to %s", address)
+	}
 	var resolvedIP string
 	if !iputil.IsIP(hostname) {
 		resolvedIP = c.dialer.GetDialedIP(hostname)

--- a/pkg/tlsx/ztls/ztls.go
+++ b/pkg/tlsx/ztls/ztls.go
@@ -129,6 +129,9 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 	if err != nil {
 		return nil, errors.Wrap(err, "could not connect to address")
 	}
+	if conn == nil {
+		return nil, fmt.Errorf("could not connect to %s", address)
+	}
 	var resolvedIP string
 	if !iputil.IsIP(hostname) {
 		resolvedIP = c.dialer.GetDialedIP(hostname)

